# Problem Statement
1. **This Dataset Contain the information about the bank customer that will take loan.**
2. **Our goal is build a model that will detect weather the customer is eligible for loan or not**

# Import Packages

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os
import warnings
warnings.filterwarnings('ignore')

# Read Data
1. Data is place another folder now find the path and read data.

In [12]:
file_path=[]
for i in os.listdir("Data"):
    file_path.append(f"{os.path.join(os.getcwd())}\Data\{i}")
    print(f"{os.path.join(os.getcwd())}\Data\{i}")

C:\Users\Laptop House\Weekly Task\CreditRisk\Data\case_study1.xlsx
C:\Users\Laptop House\Weekly Task\CreditRisk\Data\case_study2.xlsx


In [24]:
df1=pd.read_excel(file_path[0])
df2=pd.read_excel(file_path[1])


KeyboardInterrupt



# Data Preprocessing
1. Top records
2. Shape of data
3. Check columns
4. Merge based on `common column`
5. Check Datatyes
6. Null values
7. Duplicates
8. Unique Values
9. Statistical Summary

# Top Records

In [29]:
df1.sample(10)

PROSPECTID  Total_TL  Tot_Closed_TL  Tot_Active_TL  \
47215       47216         2              0              2   
45759       45760         5              4              1   
6402         6403        16             11              5   
5678         5679         1              1              0   
28783       28784        10              8              2   
17617       17618         1              1              0   
19500       19501         5              0              5   
48127       48128         3              1              2   
33316       33317        10              5              5   
898           899         1              1              0   

       Total_TL_opened_L6M  Tot_TL_closed_L6M  pct_tl_open_L6M  \
47215                    0                  0            0.000   
45759                    1                  0            0.200   
6402                     2                  1            0.125   
5678                     0                  0            0.000   
28783                    0                  1            0.000   
17617                    0                  1            0.000   
19500                    2                  0            0.400   
48127                    0                  0            0.000   
33316                    1                  0            0.100   
898                      0                  0            0.000   

       pct_tl_closed_L6M  pct_active_tl  pct_closed_tl  ...  CC_TL  \
47215              0.000          1.000          0.000  ...      0   
45759              0.000          0.200          0.800  ...      0   
6402               0.062          0.312          0.688  ...      0   
5678               0.000          0.000          1.000  ...      0   
28783              0.100          0.200          0.800  ...      1   
17617              1.000          0.000          1.000  ...      0   
19500              0.000          1.000          0.000  ...      0   
48127              0.000          0.667          0.333  ...      0   
33316              0.000          0.500          0.500  ...      0   
898                0.000          0.000          1.000  ...      0   

       Consumer_TL  Gold_TL  Home_TL  PL_TL  Secured_TL  Unsecured_TL  \
47215            1        0        0      0           0             2   
45759            0        4        0      0           4             1   
6402             0       11        0      0          16             0   
5678             0        0        0      0           1             0   
28783            6        0        0      0           2             8   
17617            1        0        0      0           0             1   
19500            4        0        0      0           0             5   
48127            0        0        0      1           1             2   
33316            5        4        0      1           4             6   
898              0        0        0      0           1             0   

       Other_TL  Age_Oldest_TL  Age_Newest_TL  
47215         1             10             10  
45759         1             40              4  
6402          5             67              1  
5678          0             64             64  
28783         1            137             10  
17617         0             15             15  
19500         1             10              2  
48127         1             39              9  
33316         0            109              4  
898           0             48             48  

[10 rows x 26 columns]

In [28]:
df2.sample(10)

PROSPECTID  time_since_recent_payment  time_since_first_deliquency  \
42672       42673                         19                       -99999   
13723       13724                         62                       -99999   
15919       15920                       1316                           34   
39554       39555                         59                           21   
47645       47646                        818                       -99999   
42210       42211                        103                            7   
16433       16434                         47                       -99999   
39957       39958                         90                       -99999   
25081       25082                         62                       -99999   
1204         1205                         43                       -99999   

       time_since_recent_deliquency  num_times_delinquent  \
42672                        -99999                     0   
13723                        -99999                     0   
15919                            34                     1   
39554                            20                     2   
47645                        -99999                     0   
42210                             3                     4   
16433                        -99999                     0   
39957                        -99999                     0   
25081                        -99999                     0   
1204                         -99999                     0   

       max_delinquency_level  max_recent_level_of_deliq  num_deliq_6mts  \
42672                 -99999                          0               0   
13723                 -99999                          0               0   
15919                     10                         10               0   
39554                      6                          6               0   
47645                 -99999                          0               0   
42210                     56                         25               2   
16433                 -99999                          0               0   
39957                 -99999                          0               0   
25081                 -99999                          0               0   
1204                  -99999                          0               0   

       num_deliq_12mts  num_deliq_6_12mts  ...  pct_CC_enq_L6m_of_L12m  \
42672                0                  0  ...                     0.0   
13723                0                  0  ...                     0.0   
15919                0                  0  ...                     0.0   
39554                0                  0  ...                     0.0   
47645                0                  0  ...                     0.0   
42210                4                  2  ...                     0.0   
16433                0                  0  ...                     0.0   
39957                0                  0  ...                     0.0   
25081                0                  0  ...                     1.0   
1204                 0                  0  ...                     1.0   

       pct_PL_enq_L6m_of_ever  pct_CC_enq_L6m_of_ever  \
42672                   0.000                     0.0   
13723                   0.000                     0.0   
15919                   0.000                     0.0   
39554                   1.000                     0.0   
47645                   0.000                     0.0   
42210                   0.000                     0.0   
16433                   0.200                     0.0   
39957                   0.000                     0.0   
25081                   0.000                     1.0   
1204                    0.833                     1.0   

       max_unsec_exposure_inPct  HL_Flag  GL_Flag  last_prod_enq2  \
42672                     2.375        0        0              PL   
13723                -99999.000        0        1              HL   
15919      

# Conclussion
1. we can see that in `df2` most of the columns contain `-99999` values.
2. These `-99999` are basically `Missing value`.

# shape

In [30]:
print("Data 1 Shape: ",df1.shape)
print("Data 2 Shape: ",df2.shape)

Data 1 Shape:  (51336, 26)
Data 2 Shape:  (51336, 62)


# Conclussion 
1. Data1 contain `51336` rows and `26` Column
2. Data2 contain `51336` rows and `62` Column

In [31]:
print(df1.columns)
print(df2.columns)

Index(['PROSPECTID', 'Total_TL', 'Tot_Closed_TL', 'Tot_Active_TL',
       'Total_TL_opened_L6M', 'Tot_TL_closed_L6M', 'pct_tl_open_L6M',
       'pct_tl_closed_L6M', 'pct_active_tl', 'pct_closed_tl',
       'Total_TL_opened_L12M', 'Tot_TL_closed_L12M', 'pct_tl_open_L12M',
       'pct_tl_closed_L12M', 'Tot_Missed_Pmnt', 'Auto_TL', 'CC_TL',
       'Consumer_TL', 'Gold_TL', 'Home_TL', 'PL_TL', 'Secured_TL',
       'Unsecured_TL', 'Other_TL', 'Age_Oldest_TL', 'Age_Newest_TL'],
      dtype='object')
Index(['PROSPECTID', 'time_since_recent_payment',
       'time_since_first_deliquency', 'time_since_recent_deliquency',
       'num_times_delinquent', 'max_delinquency_level',
       'max_recent_level_of_deliq', 'num_deliq_6mts', 'num_deliq_12mts',
       'num_deliq_6_12mts', 'max_deliq_6mts', 'max_deliq_12mts',
       'num_times_30p_dpd', 'num_times_60p_dpd', 'num_std', 'num_std_6mts',
       'num_std_12mts', 'num_sub', 'num_sub_6mts', 'num_sub_12mts', 'num_dbt',
       'num_dbt_6mts', 'num_dbt_

# Conclussion
1. See that both `dataframe` one common column `PROSPECTID`.
2. We can `merge` them first and then perform other data preprocessing.

# Merge `df1` anf `df2`

In [32]:
df1.head(2)

PROSPECTID  Total_TL  Tot_Closed_TL  Tot_Active_TL  Total_TL_opened_L6M  \
0           1         5              4              1                    0   
1           2         1              0              1                    0   

   Tot_TL_closed_L6M  pct_tl_open_L6M  pct_tl_closed_L6M  pct_active_tl  \
0                  0              0.0                0.0            0.2   
1                  0              0.0                0.0            1.0   

   pct_closed_tl  ...  CC_TL  Consumer_TL  Gold_TL  Home_TL  PL_TL  \
0            0.8  ...      0            0        1        0      4   
1            0.0  ...      0            1        0        0      0   

   Secured_TL  Unsecured_TL  Other_TL  Age_Oldest_TL  Age_Newest_TL  
0           1             4         0             72             18  
1           0             1         0              7              7  

[2 rows x 26 columns]

In [33]:
df2.head(2)

PROSPECTID  time_since_recent_payment  time_since_first_deliquency  \
0           1                        549                           35   
1           2                         47                       -99999   

   time_since_recent_deliquency  num_times_delinquent  max_delinquency_level  \
0                            15                    11                     29   
1                        -99999                     0                 -99999   

   max_recent_level_of_deliq  num_deliq_6mts  num_deliq_12mts  \
0                         29               0                0   
1                          0               0                0   

   num_deliq_6_12mts  ...  pct_CC_enq_L6m_of_L12m  pct_PL_enq_L6m_of_ever  \
0                  0  ...                     0.0                     0.0   
1                  0  ...                     0.0                     0.0   

   pct_CC_enq_L6m_of_ever  max_unsec_exposure_inPct  HL_Flag  GL_Flag  \
0                     0.0                    13.333        1        0   
1                     0.0                     0.860        0        0   

   last_prod_enq2  first_prod_enq2  Credit_Score  Approved_Flag  
0              PL               PL           696             P2  
1    ConsumerLoan     ConsumerLoan           685             P2  

[2 rows x 62 columns]

In [35]:
final_df=df1.merge(df2,how='inner',on='PROSPECTID')
final_df.head(2)

PROSPECTID  Total_TL  Tot_Closed_TL  Tot_Active_TL  Total_TL_opened_L6M  \
0           1         5              4              1                    0   
1           2         1              0              1                    0   

   Tot_TL_closed_L6M  pct_tl_open_L6M  pct_tl_closed_L6M  pct_active_tl  \
0                  0              0.0                0.0            0.2   
1                  0              0.0                0.0            1.0   

   pct_closed_tl  ...  pct_CC_enq_L6m_of_L12m  pct_PL_enq_L6m_of_ever  \
0            0.8  ...                     0.0                     0.0   
1            0.0  ...                     0.0                     0.0   

   pct_CC_enq_L6m_of_ever  max_unsec_exposure_inPct  HL_Flag  GL_Flag  \
0                     0.0                    13.333        1        0   
1                     0.0                     0.860        0        0   

   last_prod_enq2  first_prod_enq2  Credit_Score  Approved_Flag  
0              PL               PL           696             P2  
1    ConsumerLoan     ConsumerLoan           685             P2  

[2 rows x 87 columns]